working version for BB

correspondMerchant opened the pull request:
This is a working version for BB data, modified based on Donna and Judith's feedback from last week.  

**main changes:** 
_`R/calculate_recommended_interventions.R`_
- changed a lot of argument names 
- rearranged arguments so the required ones are on the top 
- cost function polynomials calculations now starts at degree 0 (fixed costs)
- provided a default for grid size if not specified 
- deleted argument include_intercept

deleted _`R/get_recommended_interventions_linear_cost.R`_
the higher order function now also handles the linear cost cases 
see _`manual_tests/test_rec_int_for_BB_data.Rmd`_ for example use case when cost is linear 

please ignore _`manual_tests/PULESA.Rmd`_ for now
the next PR will focus on making it to work with PULESA.

**additional comments:**
I will work updating the overleaf file to reflect the changes made in this PR and additional comments received. Once that big table is updated, it should help clear things. 


Minh, I'd really appreciate if you could take some time to thoroughly review this PR before approving. Your feedback would be incredibly valuable, particularly:

- If any line of code or implementation choice isn't immediately clear (you can directly comment on specific lines in GitHub)
- If you'd like more context about certain function/design decisions
- If you see opportunities for improvement in the approach or implementation

No rush on this one, your review will help ensure we're writing the best possible code! And don't worry, I take criticism really well, I think constructive feedback helps us all improve.

mithu170900 commented on R/calculate_recommended_interventions.R:
```diff
@@ -1,123 +1,170 @@
-#' the center characteristics
+#'
+#' ## required arguments:
+#' @param data data.frame: input data frame.
```
Maybe a bit of rewriting to make it clearer? Before the way you wrote it data data.frame in the same line got me confused at the first glance because it thought those two are one word, not one word and its input type. My suggestions:
#' @param data A `data.frame`. The input dataset containing the variables of interest.
#' @param outcome_name A `character` string. The name of the column in the dataset that represents the outcome of interest.
#' @param outcome_type A `character` string. Specifies the type of the outcome. Must be either `"continuous"` for numeric outcomes or `"binary"` for categorical outcomes.

mithu170900 commented on R/calculate_recommended_interventions.R:
```diff
@@ -1,123 +1,170 @@
-#' the center characteristics
+#'
+#' ## required arguments:
+#' @param data data.frame: input data frame.
```
Maybe a bit of rewriting to make it clearer? Before the way you wrote it data data.frame in the same line got me confused at the first glance because it thought those two are one word, not one word and its input type. My suggestions:
#' @param data A data.frame. The input dataset containing the variables of interest.
#' @param outcome_name A character string. The name of the column in the dataset that represents the outcome of interest.
#' @param outcome_type A character string. Specifies the type of the outcome. Must be either "continuous" for numeric outcomes or "binary" for categorical outcomes.

correspondMerchant replied:
agreed

mithu170900 replied:
I mistyped. My suggestions were:
#' @param data A `data.frame`. The input dataset containing the variables of interest.
#' @param outcome_name A `character` string. The name of the column in the dataset that represents the outcome of interest.
#' @param outcome_type A `character` string. Specifies the type of the outcome. Must be either "continuous" for numeric outcomes or "binary" for categorical outcomes.
And this applies for the rest of the parameter specifications.

mithu170900 commented on R/calculate_recommended_interventions.R:
```diff
@@ -1,123 +1,170 @@
+#'
+#' ## required arguments:
+#' @param data data.frame: input data frame.
+#' @param outcome_name character: column name of the outcome of interest.
```
A question: Do we only support one outcome? Are there any code versions/ datasets that the LAGO team has done for multiple outcomes? If this is the case, should we specify that "Column name of **one** outcome of interest"?  I feel like in real life lots of users can get ambitious and want to do multiple at the same time so might as well specify it, that we deal with one at a time.

correspondMerchant replied:
Currently only one. I think we can support multiple outcome goals in later versions. PULESA actually has two outcome goals, and they are jointly optimized.

mithu170900 commented on R/calculate_recommended_interventions.R:
```diff
@@ -1,123 +1,170 @@
+#'
+#' ## optional arguments:
+#' @param glm_family character: GLM family of the intended outcome model.
+#' automatically determined if "auto":
```
I'm slightly confused with the wording, does that mean if the user does not include this parameter in the model, it is automatically determined to be either gaussian or binomial? Or do they have to specify "auto"? I think you meant the first one... Maybe something along the line of "Default value without user specification is..."

correspondMerchant replied:
yeah its the first one, will change the wording to reflect that.

mithu170900 commented on R/get_confidence_set.R:
```diff
@@ -32,14 +35,16 @@
 #'   center_characteristic_list = "birth_volume_100",
 #'   center_cha = 1.74,
-#'   outcome = BB_proportions$EBP_proportions,
+#'   outcome_data = BB_proportions$EBP_proportions,
```
I noticed that for a lot of this the users will need to specify the column name with the specific data, and it won't compile with just the column name. Perhaps in the instructions above (the comments part at the beginning), we should specify: #' @param outcome_data A `numeric` vector. This should be the column containing the outcome data, extracted directly from the dataset (e.g., `dataset$column_name` or `dataset[["column_name"]]`). Do not pass the column name as a string (e.g., `"column_name"`).

If you agree with this, we might need to create another warning message such as:
if (is.character(outcome_data)) {
    stop("`outcome_data` should be a numeric vector (e.g., dataset$column_name), not a column name as a character string.")
}
for several parameters that require parsing columns, all in one warning message.

correspondMerchant replied:
I think this example might have confused you a bit. this is just an example for testing the `get_confidence_set` function. `get_confidence_set` is not a user facing function, it is being called [here at line 489](https://github.com/correspondMerchant/LAGO-R-Package/pull/5/files#diff-69b091a2ea4ffe095b059062485d7d288fb80d86a379bacea2b114f9ff5f6cbaR489) by `calculate_recommended_interventions`. In my opinion, all the preliminary argument checks should be done in `calculate_recommended_interventions`

mithu170900 commented on R/get_recommended_interventions_higher_order_cost.R:
```diff
@@ -1,99 +1,90 @@
                                                             outcome_goal,
-                                                            outcome_goal_optimization = "numerical",
-                                                            grid_search_step_size = 0.01,
+                                                            outcome_goal_opt_method,
```
Why don't we write it out "outcome_goal_optimization_method" just to be clear? This is pretty trivial so don't worry about it if you find it unnecessary. I was reading over it and was pausing for a bit to think because I thought "opt" meant "optional" not "optimization"

correspondMerchant replied:
agreed

mithu170900 commented on R/get_recommended_interventions_higher_order_cost.R:
```diff
@@ -1,99 +1,90 @@
+#' c(10,20).
+#' @param outcome_goal numeric: the outcome goal, a desired probability or mean
+#' value.
+#' @param outcome_goal_opt_method character: the method used for calculating the
```
I assume all of these parameters are required? Is there a default for grid search or numerical? Maybe we can add a very short description (1-2 sentences) of when to use "numerical" or "grid_search" and vice versa? I think some folks from other fields might not be familiar with "grid_search" or can conceptualize it to relate to their data. For example,
@param search_method A `character` string. Specifies the method used for optimizing the model or parameters. Must be either `"grid_search"` or `"numerical"`. Default is `"numerical"`.
#'   - Use `"grid_search"` if you want to evaluate the model over a predefined set of parameter combinations. This method is useful when you have specific ranges or values to explore systematically.
#'   - Use `"numerical"` for optimization methods that rely on gradient-based or other iterative techniques to find the best parameters without exhaustively searching a predefined grid. This is generally faster and suitable for larger datasets or when the parameter space is too large for grid search.
or even shorter than this. We can make it even more relatable in the clinical trial context.

correspondMerchant replied:
agreed, it should be more clear. The default should be numerical which is passed down from the upper level. `R/get_recommended_interventions_higher_order_cost.R` is not a user facing function, but i think you brought up a good point, i will add the clarifications to the corresponding parameter in the user facing function `R/calculate_recommended_interventions.R`.

mithu170900 commented on man/calculate_recommended_interventions.Rd:
```diff
@@ -5,121 +5,170 @@
+\item{center_characteristic_list}{character vector: a vector of column names
+of the center characteristics. For example: c("characteristic1")}
+
+\item{center_characteristic_list_future}{numeric vector: a vector of values for
```
I'm sorry but I'm still confused about why we call it "future" if it's intended for facilities-level. Maybe a different way of explaining it would make it more intuitive? Or if you think the naming is fine, it'd be great if you could explain it to me directly so I can understand it from now on and follow your logic.

correspondMerchant replied:
what about `center_characteristic_list_opt` or `center_characteristic_list_optimization`? these are fixed values that need to provided for LAGO optimization to work. I know you are familiar with Daniel's paper, this would be the `Z=1.75` value that was used during optimization in his paper.

mithu170900 commented:
I have made a total of 7 comments throughout the code version. Some of them
were at the beginning, and some were in the middle so please look
carefully. If you agree with some of them then they might take quite some
time to edit. For most of the comments, I added an example of how I thought
it would work better than the original, so please let me know what you
think.

I will read the code version again tonight to see if I spot anything.

Best,
Minh

On Tue, Nov 19, 2024 at 2:49 PM correspondMerchant ***@***.***>
wrote:

> @correspondMerchant <https://github.com/correspondMerchant> requested
> your review on: #5
> <https://github.com/correspondMerchant/LAGO-R-Package/pull/5> working
> version for BB.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/correspondMerchant/LAGO-R-Package/pull/5#event-15354606667>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APPJQIYPD7AI6WO2CTAKARD2BOI4ZAVCNFSM6AAAAABSC5TM7GVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJVGM2TINRQGY3DMNY>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***
> com>
>


-- 

*Minh Bui **I*
* Boston University*

*Department of Mathematics & Statistics*

Ph.D. Candidate in Statistics

*E*: ***@***.*** ***@***.***>*

mithu170900 commented:
Anu changes to be made or should I squash and merge now?

correspondMerchant commented:
> Anu changes to be made or should I squash and merge now?

I will make all the changes that I agreed with you and update the PR, please review the PR again once its updated.

correspondMerchant commented:
Hey Minh, can you take a look at the updated code? I should have addressed all of your comments. If you have additional comments, can you use the "start review" option instead of "add single comment" option? Maybe you already used the "start review" option last time, if so, please ignore this part.

mithu170900 commented:
Everything looks good to me now. I don't have any additional comments.
